src/Microsoft.ML.Data/Data/DataViewTypeManager.cs

```diff
@@ -53,6 +53,8 @@ public static class DataViewTypeManager
         /// </summary>
         internal static DataViewType GetDataViewType(Type type, IEnumerable<Attribute> typeAttributes = null)
         {
+            //Filter attributes as we only care about TypeAttributes
+            typeAttributes = typeAttributes == null ? typeAttributes : typeAttributes.Where(attr => attr.GetType().IsSubclassOf(typeof(TypeAttribute)));
```

Comment:
>Where [](start = 86, length = 5)

Can we instead pass an `IEnumerable<DataViewTypeAttribute>`?

Reply:
We would then have to modify all callers of this function and the Knows function to do this same filtering logic as the way that they appear to be getting the list is by calling GetCustomAttributes. it seems cleaner instead to do the filtering logic here.

Reply:
Fair enough. For the `Register` method though I still think it would be better to pass the `DataViewTypeAttribute`s directly, since in all the calls I see to this method, the attributes that are passed are derived from `DataViewTypeAttribute` anyway.



---
In reply to: [322322602](https://github.com/dotnet/machinelearning/pull/4187#discussion_r322322602) [](ancestors = 322322602)

Reply:
Yeah we can change to use a dataviewtype in the register function I think. however we should leave the Knows and Get calls as is.